2.train-VAE/optimize_hyperparameters.py

```diff
@@ -0,0 +1,86 @@
+# THIS CODE WAS SOURCED FROM THE FOLLOWING URL: https://github.com/broadinstitute/cell-painting-vae/blob/master/scripts/optimize-hyperparameters.py
+
+import sys
+import pathlib
+import numpy as np
+import pandas as pd
+import keras_tuner
+
+from keras_tuner.tuners import BayesianOptimization
+import tensorflow
+from keras.callbacks import EarlyStopping 
+
+from optimize_utils import HyperVAE, CustomBayesianTunerCellPainting, CustomBayesianTunerL1000, get_optimize_args
+
+sys.path.insert(0, "../0.data-download/scripts/")
+from data_loader import load_train_test_data
+
+# Load command line arguments
+args = get_optimize_args()
+
+# Define architecture
+encoder_architecture = []
+decoder_architecture = []
+
+# Load Data
+data_directory = pathlib.Path("../0.data-download/data")
+dfs = load_train_test_data(data_directory, train_or_test = "all")
+
+train_feat = dfs[0]
+test_feat = dfs[1]
+gene_stats = dfs[2]
+
+
+# Prepare data for training
+train_features_df = train_feat.drop(columns= ["DepMap_ID", "age_and_sex"])
+test_features_df = test_feat.drop(columns= ["DepMap_ID", "age_and_sex"])
+
+# subsetting the genes 
+# create dataframe containing the 1000 genes with the largest variances and their corresponding gene label and extract the gene labels
+largest_var_df = gene_stats.nlargest(1000, "variance")
+gene_list = largest_var_df["gene_ID"].tolist()
+gene_list
+
+# create new training and testing dataframes that contain only the corresponding genes
+train_df = train_feat.filter(gene_list, axis = 1)
+test_df = test_feat.filter(gene_list, axis = 1)
+
+# scvale the data
```

Comment:
I thought we talked about removing this

Reply:
This is an important line. It filters out the 1000 genes with the highest variances that are then used for hyperparameterization

Reply:
ohhh haha you meant the line under it. I removed that normalization function just now. I didn't realize I added it to two places